File: lib/wttj_web/channels/job_socket.ex
defmodule WttjWeb.JobSocket do
  use Phoenix.Socket

  channel "jobs:all", WttjWeb.JobChannel
  @impl true
  def connect(_params, socket, _connect_info) do
    {:ok, socket}
  end

  @impl true
  def id(_socket), do: nil
end

File: assets/js/job_socket.js
import {Socket} from "phoenix"

let jobsocket = new Socket("ws://localhost:4000/socket")

jobsocket.connect()

let channel = jobsocket.channel("jobs:all", {})

channel.join()
  .receive("ok", () => { console.log("Joined successfully") })
  .receive("error", resp => { console.log("Unable to join", resp) })

channel.on("insert", (payload) => {
  console.log("Received update", payload);
});

export default jobsocket
